docs/security/access-control.md

```diff
@@ -507,6 +619,93 @@ boolean deleted =
 
+### Grant privileges to a role
+
+You can grant specific privileges to a role.
+The request path for REST API is `/api/metalakes/{metalake}/permissions/roles/{role}/{metadataObjectType}/{metadataObjectName}/grant`.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/schema/catalog1.schema1/grant
+
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/table/catalog1.schema1.table1/grant
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+
+// Grant the privilege allowing `SELEC_TABLE` for the `schema` to `role1`        
+MetadataObject schema = ...
+Role role = client.grantPrivilegesToRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));        
+
+// Grant the privilege allowing `SELEC_TABLE` for the `table` to `role1`        
+MetadataObject table = ...
+Role role = client.grantPrivilegesToRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
+```
+
+### Revoke privileges from a role
+
+You can revoke specific privileges from a role.
+The request path for REST API is `/api/metalakes/{metalake}/permissions/roles/{role}/{metadataObjectType}/{metadataObjectName}/revoke`.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/schema/catalog1.schema1/revoke
+
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/table/catalog1.schema1.table1/revoke
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`         
+MetadataObject schema = ...
+Role role = client.revokePrivilegesFromRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`         
+MetadataObject table = ...
+Role role = client.revokePrivilegesFromRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
+
```

Comment:
You may miss some </TabItem> and </Tab> here.

Reply:
I missed the comment.